src/Avalonia.Controls/VirtualizingStackPanel.cs

```diff
@@ -265,12 +265,14 @@ protected override void OnItemsChanged(IReadOnlyList<object?> items, NotifyColle
         protected override IInputElement? GetControl(NavigationDirection direction, IInputElement? from, bool wrap)
         {
             var count = Items.Count;
+            var fromControl = from as Control;
 
-            if (count == 0 || from is not Control fromControl)
+            if (count == 0 || 
+                fromControl is null && direction is not NavigationDirection.First or NavigationDirection.Last)
```

Comment:
Should this have been `direction is not (NavigationDirection.First or NavigationDirection.Last)`?

Reply:
@jnyrup answering closed / merged PRs isn't really a good idea. Make it a new discussion or issue if you have an actual issue. Or open a PR to add the proposed fix if you're certain it is an issue.